feat: add new rule to prefer single resources (#764) (#785)
zalando/restful-api-guidelines: chapters/json-guidelines.adoc

* contain only {RFC-7493}#section-2.3[unique member names] (no duplicate
  names).

[#252]

> == {SHOULD} design single resource for reading and writing

Reviewer: Resource might be ambiguous here, how about [schema](https://swagger.io/specification/#schema-object)? That seems to be what OpenAPI calls it and also where you'd usually define these (under `components/schemas`)
```suggestion
== {SHOULD} design single schema for reading and writing
```

Author: How about `resource schema`?

Reviewer: Yeah that also works!

> To simplify API specs and API maintenance, as well as the cognitive load of
> developers, endpoints should  utilize a common model for reading and writing
> the same resource type. The usualy differences between read and write models
> are best addressed by

Reviewer: ```suggestion
To simplify API specs and API maintenance, as well as the cognitive load of
developers, endpoints should utilize a common model for reading and writing
the same resource type. The differences between read and write models
are best addressed by
```

* marking properties `writeOnly` that are only available in a request, e.g.
  passwords, and
* marking properties `readOnly` that are only available in a response, e.g.
  resource identifiers.

**Note:** A resource with property marked `readOnly` in the API according to
the {json-schema-validation}#rfc.section.9.4[JSON schema validation] may be

> ignored or rejected by a server. Both are in line with the idea behind <<#109,

Reviewer 2: ```suggestion
ignored or rejected by a server. Both are in line with the idea behind 
<<#109,the compatibility guidance>>, however, we suggest to document when
rejecting to prevent client side surprises.
```

> **Note:** A resource with property marked `readOnly` in the API according to
> the {json-schema-validation}#rfc.section.9.4[JSON schema validation] may be
> ignored or rejected by a server. Both is inline with the idea behind <<#109,
> the compatibility guidance>>, however, we suggest to document rejection to
> prevent client side surprises.

Reviewer: ```suggestion
**Note:** A resource with property marked `readOnly` in the API according to
the {json-schema-validation}#rfc.section.9.4[JSON schema validation] may be
ignored or rejected by a server. Both are in line with the idea behind
<<#109, the compatibility guidance>>, however, we suggest to document
rejection to prevent client side surprises.
```
